### PR TITLE
Optimize `normalized_zero` creation in _handle_negative_values.

### PR DESCRIPTION
**Pull request by amitmarkel**
1. Create it directly on GPU instead of on CPU with memory transfers.
2. Do the negation on the in-memory item directly, not on the entire tensor after creation.

### CODE REVIEW

File: googlehydrology/utils/samplingutils.py [hunk 1]
```diff
@@ -128,7 +128,9 @@ def _handle_negative_values(
         .sel(parameter='scale')
         .item()
     )
-    normalized_zero = -torch.tensor(center / scale).to(values)
+    normalized_zero = torch.tensor(
+        -center / scale, device=values.device, dtype=values.dtype
```
Comment (omrishefi) [inline googlehydrology/utils/samplingutils.py]: I think writing -(center/scale) is more readable

**Review by omrishefi — DISMISSED** [review]

File: googlehydrology/utils/samplingutils.py (hunk 1 above)
Comment (amitmarkel) [reply googlehydrology/utils/samplingutils.py]: Done.

**Comment by amitmarkel** [comment]
(need reapproval due to merging with main)

**Review by omrishefi — APPROVED** [review]
